rust/routee-compass/src/app/compass/response/response_writer.rs

```diff
@@ -0,0 +1,78 @@
+use super::response_output_format::ResponseOutputFormat;
+use crate::app::compass::compass_app_error::CompassAppError;
+use std::io::prelude::*;
+use std::{
+    fs::File,
+    sync::{Arc, Mutex},
+};
+
+pub enum ResponseWriter {
+    NoOutputWriter,
+    RunBatchFileSink {
+        filename: String,
+        file: Arc<Mutex<File>>,
+        format: ResponseOutputFormat,
+        delimiter: Option<String>,
+    },
+}
+
+impl ResponseWriter {
+    pub fn write_response(&self, response: &serde_json::Value) -> Result<(), CompassAppError> {
+        match self {
+            ResponseWriter::NoOutputWriter => Ok(()),
+            ResponseWriter::RunBatchFileSink {
+                filename: _,
+                file,
+                format,
+                delimiter: _,
+            } => {
+                let file_ref = Arc::clone(file);
+                let mut file_attained = file_ref.lock().map_err(|e| {
+                    CompassAppError::ReadOnlyPoisonError(format!(
+                        "Could not aquire lock on output file: {}",
+                        e
+                    ))
+                })?;
+
+                // if write_delimiter {
```

Comment:
noting, this solution doesn't really give us the ability to write a delimiter, since the write can be called from different threads and we cannot know when we are between rows. to get there, i would want to wrap the writer itself in a mutex, hold a buffer, and then apply a delimiter when appropriate. but since reproducing our current capability was the goal, this is what we've got.